packages/cardinal-integration/src/cardinal-three-d-secure-flow-v2.ts

```diff
@@ -0,0 +1,127 @@
+import { merge, some } from 'lodash';
+
+import {
+    HostedForm,
+    isCreditCardInstrument,
+    isVaultedInstrument,
+    OrderPaymentRequestBody,
+    OrderRequestBody,
+    PaymentIntegrationSelectors,
+    PaymentIntegrationService,
+    PaymentMethod,
+    PaymentRequestOptions,
+    PaymentStrategy,
+    RequestError,
+} from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+import { CardinalThreeDSecureToken } from './cardinal';
+import CardinalClient, { CardinalOrderData } from './cardinal-client';
+
+export default class CardinalThreeDSecureFlowV2 {
+    constructor(
+        private _paymentIntegrationService: PaymentIntegrationService,
+        private _cardinalClient: CardinalClient,
+    ) {}
+
+    async prepare(method: PaymentMethod): Promise<void> {
+        await this._cardinalClient.load(method.id, method.config.testMode);
+    }
+
+    async start(
+        execute: PaymentStrategy['execute'],
+        payload: OrderRequestBody,
+        options?: PaymentRequestOptions,
+        hostedForm?: HostedForm,
+    ): Promise<void> {
+        const { getCardInstrument } = this._paymentIntegrationService.getState();
+        const { payment = { methodId: '' } } = payload;
+        const { paymentData = {} } = payment;
+
+        try {
+            return await execute(payload, options);
+        } catch (error) {
+            if (
+                error instanceof RequestError &&
+                error.body.status === 'additional_action_required'
+            ) {
+                const token = error.body.additional_action_required?.data?.token;
+                const xid = error.body.three_ds_result?.payer_auth_request;
+
+                await this._cardinalClient.configure(token);
+
+                const bin = this._getBin(paymentData, getCardInstrument, hostedForm);
+
+                if (bin) {
+                    await this._cardinalClient.runBinProcess(bin);
+                }
+
+                try {
+                    return await this._submitPayment(payment, { xid }, hostedForm);
+                    // eslint-disable-next-line @typescript-eslint/no-shadow
+                } catch (error) {
```

Comment:
To avoid the above eslint rule, you should rename this variable to another. For example, it can be `err`.

Reply:
Fixed